### PR TITLE
[ANDI-8996/AnAm]: Added prototype to parse VLAN encapsulated TECMP pa…

### PR DESCRIPTION
**Pull request by aamereller**
…ckets.

### CODE REVIEW

File: src/tecmp.c [hunk 1]
```diff
@@ -53,6 +53,10 @@ int tecmp_next(
 	uint16_t ether = ntoh16(*((uint16_t*)(raw_frame + 12)));
 
 	// Ethertype mismatch
```
Comment (kayoub5) [inline src/tecmp.c:55]: Make sure this comment stays next to the `if (!(ether == 0x2090 || ether == 0x99FE)) {` line

File: src/tecmp.c [hunk 2]
```diff
@@ -53,6 +53,10 @@ int tecmp_next(
...
+	if (ether == 0x8100) {
+		uint32_t skip = ETH_SIZE + 2;
+		return tecmp_nect(raw_frame + skip, raw_frame_size - skip, iterator, header, data);
+	}
```
Comment (kayoub5) [inline src/tecmp.c:59]: Packet can have up to 2 VLANS, Ethertype could be one of 0x8100, 0x88a8 or 0x9100

File: src/tecmp.c [hunk 3]
```diff
@@ -52,6 +52,11 @@ int tecmp_next(
 
 	uint16_t ether = ntoh16(*((uint16_t*)(raw_frame + 12)));
 
+	// VLAN ethertypes
+	if ((ether == 0x8100) || (ethertype == 0x88A8) || (ethertype == 0x9100)) {
+		uint32_t skip = ETH_SIZE + 2;
+		return tecmp_nect(raw_frame + skip, raw_frame_size - skip, iterator, header, data);
```
Comment (kayoub5) [inline src/tecmp.c:58]: typo, tecmp_next

**Comment by kayoub5** [comment]
@aamereller PR fails to compile

File: CMakeLists.txt [hunk 4]
```diff
@@ -2,12 +2,12 @@ cmake_minimum_required(VERSION 3.17)
...
 # Project name and a few useful settings. Other commands can pick up the results
 project(LIBTECMP
-    VERSION 0.1
+    VERSION 0.2
     DESCRIPTION "TECMP Library"
 )
 
 # Only do these if this is the main project, and not if it is included through add_subdirectory
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)#
```
Comment (kayoub5) [inline CMakeLists.txt:10]:
```suggestion
if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

**Comment by aamereller** [comment]
Test has been manually executed using the converter mentioned above. Converter execution did not show errors, created a (seemingly) valid pcap, and returned with exit status 0.
Thus I consider the library tested and verified.

File: src/tecmp.c [hunk 5]
```diff
@@ -52,8 +52,14 @@ int tecmp_next(
...
 	uint16_t ether = ntoh16(*((uint16_t*)(raw_frame + 12)));
 
+	// VLAN ethertypes
+	if ((ether == 0x8100) || (ether == 0x88A8) || (ether == 0x9100)) {
+		uint32_t skip = 4;
+		return tecmp_next(raw_frame + skip, raw_frame_size - skip, iterator, header, data);
+	}
 	// Ethertype mismatch
 	if (!(ether == 0x2090 || ether == 0x99FE)) {
+		std::cout<<"Ether type "<<ether<<" is invalid."<<std::endl;
```
Comment (kayoub5) [inline src/tecmp.c:62]: Remove debug

**Comment by kayoub5** [comment]
@aamereller please merge latest changes from master into your branch

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**